### ISSUE TITLE
Test suite fails on 32 bits

### ISSUE THREAD

**Issue by videlec** [videlec/e-antic#61]
As reported by Jerome Benoit, there are two test failures on 32 bits
- `poly_extra/test/t-fmpz_poly_isolate_real_roots` (expected to be solved by #62)
- `renf/test/t-init_nth_root_fmpq`
```
FAIL: poly_extra/test/t-fmpz_poly_isolate_real_roots
====================================================

a  = 598995133/1776349264
r1 = -8
r2 = -4
Failed root1 of poly1
FAIL poly_extra/test/t-fmpz_poly_isolate_real_roots (exit status: 1)

FAIL: renf/test/t-init_nth_root_fmpq
====================================

FAIL renf/test/t-init_nth_root_fmpq (exit status: 134)
```
The compilation warnings already gives a hint on what went wrong for one of them
```
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c: In function 'main':
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:82:20: warning: overflow in conversion from 'long long int' to 'mp_limb_signed_t' {aka 'long int'} changes value from '-2497397708202708803' to '598995133' [-Woverflow]
     fmpq_set_si(a, WORD(-2497397708202708803), WORD(623751320421068880)); /* approx of root 1 */
                    ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:82:48: warning: unsigned conversion from 'long long int' to 'mp_limb_t' {aka 'long unsigned int'} changes value from '623751320421068880' to '1776349264' [-Woverflow]
     fmpq_set_si(a, WORD(-2497397708202708803), WORD(623751320421068880)); /* approx of root 1 */
                                                ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:89:20: warning: overflow in conversion from 'long long int' to 'mp_limb_signed_t' {aka 'long int'} changes value from '-83382880964438876' to '667161764' [-Woverflow]
     fmpq_set_si(a, WORD(-83382880964438876), WORD(172646812034399449)); /* approx of root 2 */
                    ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:89:46: warning: unsigned conversion from 'long long int' to 'mp_limb_t' {aka 'long unsigned int'} changes value from '172646812034399449' to '1592392921' [-Woverflow]
     fmpq_set_si(a, WORD(-83382880964438876), WORD(172646812034399449)); /* approx of root 2 */
                                              ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:112:20: warning: overflow in conversion from 'long long int' to 'mp_limb_signed_t' {aka 'long int'} changes value from '-1729553040654133611' to '1335694997' [-Woverflow]
     fmpq_set_si(a, WORD(-1729553040654133611), WORD(995268582037528766));
                    ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:112:48: warning: unsigned conversion from 'long long int' to 'mp_limb_t' {aka 'long unsigned int'} changes value from '995268582037528766' to '2423529662' [-Woverflow]
     fmpq_set_si(a, WORD(-1729553040654133611), WORD(995268582037528766));
                                                ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:118:20: warning: overflow in conversion from 'long long int' to 'mp_limb_signed_t' {aka 'long int'} changes value from '1148746836908268129' to '1858427489' [-Woverflow]
     fmpq_set_si(a, WORD(1148746836908268129), WORD(150128881916767));
                    ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:118:47: warning: unsigned conversion from 'long long int' to 'mp_limb_t' {aka 'long unsigned int'} changes value from '150128881916767' to '2595052383' [-Woverflow]
     fmpq_set_si(a, WORD(1148746836908268129), WORD(150128881916767));
                                               ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:141:20: warning: overflow in conversion from 'long long int' to 'mp_limb_signed_t' {aka 'long int'} changes value from '606001226841264272' to '-1286401904' [-Woverflow]
     fmpq_set_si(a, WORD(606001226841264272), WORD(806515533049393));
                    ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:141:46: warning: unsigned conversion from 'long long int' to 'mp_limb_t' {aka 'long unsigned int'} changes value from '806515533049393' to '2279239217' [-Woverflow]
     fmpq_set_si(a, WORD(606001226841264272), WORD(806515533049393));
                                              ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:147:20: warning: overflow in conversion from 'long long int' to 'mp_limb_signed_t' {aka 'long int'} changes value from '607804650398072193' to '-137446015' [-Woverflow]
     fmpq_set_si(a, WORD(607804650398072193), WORD(806515533049393));
                    ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:147:46: warning: unsigned conversion from 'long long int' to 'mp_limb_t' {aka 'long unsigned int'} changes value from '806515533049393' to '2279239217' [-Woverflow]
     fmpq_set_si(a, WORD(607804650398072193), WORD(806515533049393));
                                              ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:170:20: warning: overflow in conversion from 'long long int' to 'mp_limb_signed_t' {aka 'long int'} changes value from '370174771560547' to '130252899' [-Woverflow]
     fmpq_set_si(a, WORD(370174771560547), WORD(804552393201742478));
                    ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:170:43: warning: unsigned conversion from 'long long int' to 'mp_limb_t' {aka 'long unsigned int'} changes value from '804552393201742478' to '2415587982' [-Woverflow]
     fmpq_set_si(a, WORD(370174771560547), WORD(804552393201742478));
                                           ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:176:20: warning: overflow in conversion from 'long long int' to 'mp_limb_signed_t' {aka 'long int'} changes value from '442898231808688' to '1204245168' [-Woverflow]
     fmpq_set_si(a, WORD(442898231808688), WORD(962580472320952871));
                    ^~~~
../poly_extra/test/t-fmpz_poly_isolate_real_roots.c:176:43: warning: unsigned conversion from 'long long int' to 'mp_limb_t' {aka 'long unsigned int'} changes value from '962580472320952871' to '4033546791' [-Woverflow]
     fmpq_set_si(a, WORD(442898231808688), WORD(962580472320952871));
```

**Comment by jgmbenoit** [comment]
For the second issue related to renf/test/t-init_nth_root_fmpq  ,
I have just noticed a couple of things:
1] the test is not that random: d always take the same values;
2] the same values are architecture dependant;
3] the test fail on i386 arch when d has a negative value.

**Comment by jgmbenoit** [comment]
Concerning renf/test/t-init_nth_root_fmpq , if d is imposed to be positive, then the test passed.
So, I guess that this test failure reveals in fact an issue with n_randtest_not_zero

**Comment by videlec** [comment]
@jgmbenoit Thanks for your investigation! Indeed, `d` is supposed to be positive. I will add a fix in the following days.

**Comment by videlec** [comment]
The rest should be fixed in #63
